lib/logstash/inputs/stomp.rb

```diff
@@ -30,26 +30,42 @@ class LogStash::Inputs::Stomp < LogStash::Inputs::Base
   # The vhost to use
   config :vhost, :validate => :string, :default => nil
 
+  # Auto reconnect
+  config :reconnect, :validate => :boolean, :default => true
+  
+  #Auto reconnect interval in seconds
+  config :reconnect_interval, :validate => :number, :default => 30
+
   # Enable debugging output?
   config :debug, :validate => :boolean, :default => false
 
   private
   def connect
     begin
       @client.connect
-      @logger.debug? && @logger.debug("Connected to stomp server") if @client.connected?
-    rescue OnStomp::ConnectFailedError, OnStomp::UnsupportedProtocolVersionError=> e
-      @logger.warn("Failed to connect to stomp server, will retry", :exception => e, :backtrace => e.backtrace)
-      if stop?
-        sleep 2
+      @logger.info("Connected to stomp server") if @client.connected?
+    rescue OnStomp::ConnectFailedError, OnStomp::UnsupportedProtocolVersionError, Errno::ECONNREFUSED => e      
+      if @reconnect
+	@logger.warn("Failed to connect to stomp server. Retry in #{@reconnect_interval} seconds. #{e.inspect}")
+	@logger.debug("#{e.backtrace.join("\n")}")
+	sleep @reconnect_interval
         retry
       end
+      @logger.warn("Failed to connect to stomp server. Exiting with error: #{e.inspect}")
+      @logger.debug("#{e.backtrace.join("\n")}")
+      stop?
     end
   end
 
   public
   def register
     require "onstomp"
+    require "logger"
```

Comment:
this is not necessary nor wanted. Logstash plugin inherit the @logger from its base class, 
I suppose you used it for testing but should not be merged in
can you please remove this logger initialisation, please